src/api/app/controllers/staging_workflow_controller.rb

```diff
@@ -0,0 +1,14 @@
+class StagingWorkflowController < ApplicationController
+  before_action :set_project
+
+  def staged_requests
+    @requests = @project.staged_requests
```

Comment:
We are in the process of renaming all custom actions to "CRUD" like routes and moving controller code to seperate controllers where needed (basically following [this pattern](http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/).

I would vouch for doing this here as well.

Reply:
yes, if you see in the API controllers, derived from ```source_controller```, I did it already, however in this case, the results didn't really look appealing to me. I will give an example:

as I did we have the following route:

```/staging_workflow/:project/staged-requests``` 

which I personally like. 

Using the namespace like that:

```
  namespace :staging_workflow do
    resources :staged_requests, only: :show, param: :project
  end
```

generates an ugly url like: 

```/staging_workflow/staged_requests/:project```

Any suggestion? Otherwise I would like to keep as I did it, even because we don't know if its final, if our customer want to have it and so on...

Reply:
I find this discussion and the dicussion in 6109 a little disconnected, because:
once :project next to staging_workflow refers to the project the staging workflow is of and another time :project refers to the staging project. I find this mildly confusing

Reply:
Agree with coolo, we should look at this from the "big picture"

Also structuring the controllers like @bgeuken suggested does not imply that you need to structure the routes like this as well. You can do 
```
/staging_workflow/requests/show
```

Reply:
@coolo yup, but I think it's normal since there was no discussion about how the urls should look like, specially for this single endpoint api route, and in my case, the code was already ready since #6071

Reply:
I was referring to the naming of the controller code only, like Christian said.

Reply:
> Agree with coolo, we should look at this from the "big picture"

which is? 

> Also structuring the controllers like @bgeuken suggested does not imply that you need to 
> structure the routes like this as well. You can do

nothing imply, but ```request#show``` is IMO wrong, it should be the ``request#index´´´, and  we want to show just the staged requests, and not all, and yes we could use query_string, but then again, if there is no "big picture"  it's maybe not what we want..

Reply:
It would of course be index:-)

Reply:
@bgeuken yup, but @ChrisBr said something differently.. the answer was for him

Reply:
> but @ChrisBr said something differently.. 

my comment was meant as example ....

> which is?

Looking at the workflow and see what we actually need and agree on it and not everyone working in silos and discussing it in every PR :thinking:

Reply:
> > but @ChrisBr said something differently..
> 
> my comment was meant as example ....

Hm @ChrisBr if you are asking for changes, providing "wrong" examples, do more harm than good :/

> Looking at the workflow and see what we actually need and agree on it and not everyone >working in silos and discussing it in every PR

Yup, I looked in the workflow, asked in the sprint planning meeting and I'm delivering what is asked in the card. Beside you can reach me via all channels that we have, daily scrum, irc, gotomeeting, mumble, mobile, you name it.. but here you are discussing in the PR..